### ISSUE TITLE
Allow form element default values to be set from URL parameters

### ISSUE THREAD

**Issue by mattwomple** [ampproject/amphtml#24533]
## Describe the new feature or change to an existing feature you'd like to see

When a user navigates to a search results URL like `/search?q=my+search`, it is typically expected that the search text input on the page is populated with the search term, in this case `my search`. It is already possible to populate an `<amp-list>` via query parameters thanks to AMP variable substitution with `QUERY_PARAM`, so it would be great if AMP allowed setting default values of form inputs by checking the URL. For example, if new `<amp-form>` attribute `initialize-from-url` existed, then inputs under the form could have their default value set from the URL GET parameters.

A minimal example of this kind of search page is available at [android.cmphys.com/temp/amp-form-fr.html?q=my+search](https://android.cmphys.com/temp/amp-form-fr.html?q=my+search). Notice results populate but the input on the page is blank on page load.

## Describe alternatives you've considered

The workaround I currently use is to put each form input under an `amp-list` and query a special endpoint I created to echo back query parameters. Example: `https://ampsupport.wompmobile.com/echoquery/dictionaryformat?q=QUERY_PARAM(q)`. This has obvious disadvantages:

- Server call necessary just to render an input
- `placeholder` and `fallback` in `amp-list` cannot contain the same input or else the form risks having multiple inputs with the same `name` attribute (they are all in the DOM even if they are not all visible)

You can see the minimal example with this workaround incorporated at [android.cmphys.com/temp/amp-form-ex.html?q=my+search](https://android.cmphys.com/temp/amp-form-ex.html?q=my+search).

## Additional context

While related feature request #14637 would also be nice to have, it would not resolve this issue since `amp-bind` does not apply states on page load.

**Comment by mattwomple** [comment]
@ampproject/wg-ui-and-a11y I have some initial work for this feature request: https://github.com/mattwomple/amphtml/commit/e0252509d2e7919f2fd5694d258c04c758a1426b . This commit enables the feature requested, but obviously lacks documentation and tests. Before I venture further, I would appreciate feedback about whether this feature would be considered for inclusion.

**Comment by mattwomple** [comment]
Deeper searching reveals this is a duplicate feature request: #7600 and #13823

Still, there is some additional information in this thread, including a (clumsy) workaround and a proof-of-concept commit #24671.

CC @caroqliu since I see you were recently assigned to #13823.

**Comment by caroqliu** [comment]
Hi there! Thank you for taking the time to investigate this issue and develop a workaround. I will take a look at this today.

**Comment by choumx** [comment]
Sounds reasonable though latency could be annoying and it may only work on publisher origin (non-cache) AMP pages. It sounds like that's your intended use case though, right?

Let's also de-dupe to have a single conversation thread.
